datastore/accountmanager.go

```diff
@@ -84,6 +84,17 @@ const listNotUserAccountsSQL = `
 	)
 `
 
+const syncUpsertAccountSQL = `
+	WITH upsert AS (
+		update account set authority_id=$2, assertion=$3, resellerapi=$4
+		where id=$1
+		RETURNING *
+	)
+	insert into account (id,authority_id,assertion,resellerapi)
+	select $1, $2, $3, $4
+	where not exists (select * from upsert)
```

Comment:
Have you tried on conflict do update sentence. I think it reduces a bit the code

Reply:
I think that's only available at later postgresql versions, so I've avoided it deliberately... until the world moves on.